contracts/utils/math/Math.sol

```diff
@@ -33,8 +55,25 @@ library Math {
      */
     function trySub(uint256 a, uint256 b) internal pure returns (bool success, uint256 result) {
         unchecked {
-            if (b > a) return (false, 0);
-            return (true, a - b);
+            success = a >= b;
+            // equivalent to: success ? (a - b) : 0
+            result = SafeCast.toUint(success) * (a - b);
+        }
+    }
+
+    /**
+     * @dev Unsigned saturating multiplication, bounds to `2 ** 256 - 1` instead of overflowing.
+     */
+    function saturatingMul(uint256 a, uint256 b) internal pure returns (uint256) {
```

Comment:
use of assembly may be efficient, but readability is not great. I'd be more confortable with that function being

```solidity
(bool success, uint256 result) = tryMul(a, b);
return ternary(success, result, type(uint256).max);
```

Reply:
Yeah I had to use assembly because `c / a` reverts, even inside a `unchecked` statement, while in pure evm a division by zero always result in zero.

Reply:
Btw I opened an issue in the solidity compiler, division should not revert in unchecked blocks:
https://github.com/ethereum/solidity/issues/15200